stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/security/KeyStore.java

```diff
@@ -0,0 +1,244 @@
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.security.KeyStore.PasswordProtection;
+import java.security.KeyStore.PrivateKeyEntry;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.Certificate;
+
+/**
+ * KeyStore process the configured keystore and trust store and provide an API.
+ *
+ * @since 0.0.963.1
+ */
+public class KeyStore {
+
+    private static final Logger log = LoggerFactory.getLogger(KeyStore.class);
+
+    // KeyStore configurations
+    private static final String KEY_STORE_CONFIG = "keyStore";
+    private static final String KEY_STORE_LOCATION = "location";
+    private static final String KEY_STORE_TYPE = "type";
+    private static final String KEY_STORE_PASSWORD = "keyStorePassword";
+    private static final String KEY_ALIAS = "keyAlias";
+    private static final String KEY_PASSWORD = "keyPassword";
+    // TrustStore configurations
+    private static final String TRUST_STORE_CONFIG = "trustStore";
+    private static final String TRUST_STORE_LOCATION = "location";
+    private static final String TRUST_STORE_TYPE = "type";
+    private static final String TRUST_STORE_PASSWORD = "trustStorePassword";
+
+    private static final KeyStore keyStoreInstance = new KeyStore();
+    private java.security.KeyStore keyStore;
+    private java.security.KeyStore trustStore;
+
+    private KeyStore() {
+        loadKeyStore();
+        loadTrustStore();
+    }
+
+    /**
+     * Get the KeyStore.
+     *
+     * @return KeyStore instance
+     */
+    public static KeyStore getKeyStore() {
+        return keyStoreInstance;
+    }
+
+    /**
+     * Get the public key for a given key alias from trustStore.
+     *
+     * @param alias
+     * @return public key corresponding to the alias.
+     * @throws KeyStoreException
+     */
+    public PublicKey getTrustedPublicKey(String alias) throws KeyStoreException {
+        Certificate certificate = getTrustedCertificate(alias);
+        if (certificate == null) {
+            return null;
+        }
+        return (PublicKey) certificate.getPublicKey();
+    }
+
+    /**
+     * Get the certificate for a given key alias from trustStore.
+     *
+     * @param alias
+     * @return certificate corresponding to the alias.
+     * @throws KeyStoreException
+     */
+    public Certificate getTrustedCertificate(String alias) throws KeyStoreException {
+        try {
+            return trustStore.getCertificate(alias);
+        } catch (java.security.KeyStoreException e) {
+            throw new KeyStoreException("Failed to load certificate: " + alias, e);
+        }
+    }
+
+    /**
+     * Get the private key for a given key alias.
+     *
+     * @param alias
+     * @param keyPassword
+     * @return private key corresponding to the alias
+     * @throws KeyStoreException
```

Comment:
Descriptions of the parameters/exceptions are missing. Complete the method comments in all related places.